core/src/test/java/org/kiwix/kiwixmobile/core/dao/HistoryDaoTestUtil.kt

```diff
@@ -0,0 +1,19 @@
+package org.kiwix.kiwixmobile.core.dao
+
+import io.mockk.every
+import io.mockk.mockk
+import org.kiwix.kiwixmobile.core.history.HistoryListItem
+
+fun mockkHistoryItemWithTitle(historyTitle: String): HistoryListItem.HistoryItem {
+  val historyItem = mockk<HistoryListItem.HistoryItem>()
+  every { historyItem.historyUrl } returns "url$historyTitle"
+  every { historyItem.dateString } returns "2012-01-01"
+  every { historyItem.databaseId } returns 0
+  every { historyItem.zimId } returns "zimId"
+  every { historyItem.zimName } returns "zimName$historyTitle"
+  every { historyItem.zimFilePath } returns "zimPath$historyTitle"
+  every { historyItem.favicon } returns "favicon"
+  every { historyItem.historyTitle } returns historyTitle
+  every { historyItem.timeStamp } returns 0
+  return historyItem
```

Comment:
Instead of mocking every field of `HistoryItem` can we follow the recommendation [here](https://phauer.com/2018/best-practices-unit-testing-kotlin/#data-classes-for-assertions) to use the fact that `HistoryItem` is a data class?

Reply:
Missed that, thanks!

Reply:
I ran into a problem. The pre-inserted historyItem has a constant, databaseID = 0 before insertion. When the historyItem is inserted into the database it is acutally copied with a new databaseID != 0.

(My thinking is: ) That means I'd have to mock the pre-inserted historyItem and change it's databaseId to 1. As mocked objects does not have an implemented equals method, the assertEquals method always fails when using the mocked historyItem. If I can't solve this without a clean solution I think the original solution is fine. 

Here is my current wip implementation for one of the tests with the output from the failing test:

```
val historyItem = HistoryItem("url",
  "string",
  "2012-02-01",
  199002,
  mockkZimFileReader())

// Creating a spy to change the databaseId while keeping the fields from historyItem
val mockedItem: HistoryItem = spyk(historyItem)
historyDao.saveHistory(mockedItem)
every { mockedItem.databaseId } returns 1

val historyList: List<HistoryItem> = historyDao
  .getHistoryList(false, "")

// This fails with the output:
// Expected :HistoryItem(#4)
// Actual   :HistoryItem(databaseId=1, zimId=1, zimName=zimName, zimFilePath=path, //favicon=favicon, historyUrl=url, historyTitle=string, dateString=2012-02-01, //timeStamp=199002, id=1)
assertEquals(mockedItem, historyList[0])

```

Reply:
> The pre-inserted historyItem has a constant, databaseID = 0 before insertion. When the historyItem is inserted into the database it is acutally copied with a new databaseID != 0.

:confused: Are you sure? Seems weird. Maybe I am not familiar with objectbox.

Reply:
Sorry, I missed the fact that `databaseId` is a `val` and can't be reassigned. I got it to work for `NewBookDaoTest` by setting `databaseId` to 1 in the constructor itself. You can check #1870

Reply:
@srv-twry Hmm, HistoryItem does not have a databaseID in any of its constructors. Do you know how the Box changes the id of the historyentity?